apps/relayer/README.md

```diff
@@ -1,5 +1,146 @@
-# Relayer service
+# MACI Relayer Service
 
-## Instructions
+This repository contains the setup and instructions to run the MACI Relayer service along with MongoDB using Docker.
 
-1. Add `.env` file (see `.env.example`).
+## Flow Diagram
+
+Below is a high-level diagram illustrating the flow of the Relayer Service:
+
+![MACI Relayer Flow](./relayer-diagram.png)
+
+## High-Level Overview of the Relayer:
+
+### Functionality
+
+The Relayer serves as an intermediary between users and the MACI system. It performs the following tasks:
+
+- Allows users to send messages without having to pay gas fees by processing these messages off-chain.
+- Receives user messages, such as votes or sign-ups.
+- Prepare messages batch and upload it to IPFS
+- Submits the message hashes to the Poll contract with IPFS hash of the batch.
+
+This process ensures that user actions are recorded on-chain without compromising their privacy.
+
+### Privacy Preservation
+
+By handling the creation and submission of zero-knowledge proofs, the Relayer ensures that sensitive user information remains confidential. This design choice is essential for maintaining the integrity and trustworthiness of the voting process.
+
+### Decentralization and Trust
+
+The Relayer can be self-hosted, which enhances privacy guarantees and reduces reliance on a single trusted party. This decentralization aligns with the broader goals of the MACI framework to minimize trust assumptions and potential points of failure.
```

Comment:
I'm not sure this is true,  the service is run by the coordinator/admin so not really decentralised - and currently allowing others to run it and add them as relayers on the system would open up to malicious activity as they could publish invalid batch hashes and dos the round

Reply:
I think it's true for the party that all the users trust. The same as for coordinator but it can be a different 3rd party.

Reply:
but that does not minimise points of failure, it would actually increase them

Reply:
I added a comment for this block that relayer must be trusted by users and coordinator.

Reply:
I still don't think that's a true statement unfortunately.
Also: "Anyway, relayer must be trusted by coordinator and users due to current status of MACI design." it's very vague and doesn't really give an explanation 

My suggestion is to remove this part, and mention that it's recommended that the coordinator only runs this service for now. That is because if the service malfunctions and batch hashes are pushed on chain without the messages actually being stored on ipfs, then rounds are automatically botched